### PR TITLE
Add ImageMagick

### PR DESCRIPTION
**Pull request by SimonDanisch**
Probably won't work right away?!
Swapped libjpeg with JpegTurbo...

### CODE REVIEW

File: I/ImageMagick/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    done
+fi
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products(prefix) = [
```
Comment (giordano) [inline I/ImageMagick/build_tarballs.jl]: `products` is not an array, not a function, just remove `prefix` everywhere

File: I/ImageMagick/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    LibraryProduct(prefix, "libMagickWand", :libwand),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    "Zlib_jll",
+    "Libpng_jll",
```
Comment (giordano) [inline I/ImageMagick/build_tarballs.jl]: It's currently named `libpng_jll`

File: I/ImageMagick/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+# Collection of sources required to build imagemagick
+sources = [
+    "https://github.com/ImageMagick/ImageMagick6/archive/6.9.10-12.tar.gz" =>
+    "efaae51489af9f895762bcb7090636f03194daaa026eda97dae230098d2ccec7",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/ImageMagick6*/
+if [[ ${target} == *-apple-* ]]; then
```
Comment (giordano) [inline I/ImageMagick/build_tarballs.jl]: This work around won't probably be needed, but I can't check right now

**Comment by staticfloat** [comment]
I merged this; here is the ["upgrade" commit](https://github.com/JuliaPackaging/Yggdrasil/commit/77d12ef9d7e7a823e5d3eddd4dd8b21270824090) to get this to work with the next version of BinaryBuilder (other than just changing the dependencies to use JLL package names).  Note that we can get rid of a lot of the special-casing because BB will take care of things automatically for us

**Comment by SimonDanisch** [comment]
Love it :)

**Pull request merged**